src/tools/fuzzing.h

```diff
@@ -783,10 +898,81 @@ class TranslateToFuzzReader {
     ReFinalize().walkFunctionInModule(func, &wasm);
   }
 
+  void modifyInitialFunctions() {
+    if (wasm.functions.empty()) {
+      return;
+    }
+    // Pick a chance to fuzz the contents of a function.
+    const int RESOLUTION = 10;
+    auto chance = upTo(RESOLUTION + 1);
+    for (auto& ref : wasm.functions) {
+      auto* func = ref.get();
+      prepareToCreateFunctionContents(func);
+      if (func->imported()) {
+        // We can't allow extra imports, as the fuzzing infrastructure wouldn't
+        // know what to provide.
+        func->module = func->base = Name();
+        func->body = make(func->sig.results);
+      }
+      // Optionally, fuzz the function contents.
+      if (upTo(RESOLUTION) >= chance) {
+        dropToLog(func);
+        // TODO add some locals? and the rest of addFunction's operations?
+        // TODO: interposition, replace initial a(b) with a(RANDOM_THING(b))
+        recombine(func);
+        mutate(func);
+        fixLabels(func);
```

Comment:
It [seems](https://github.com/WebAssembly/binaryen/blob/2dc156d1d3af7f3eb40bdce7c20aa84c98d50c15/src/tools/fuzzing.h#L548-L559) we do these transformations only when we allow OOB. Is it OK not to check that here too?

Reply:
It would not help much to check that here. When we generate code ourselves, we can add OOB checks into it as we create it. So it makes sense to check there. But for initial contents, we didn't create them and they don't have OOB checks anyhow, so it's likely they can have OOB which will trap (this is a downside of initial contents, atm). But it's still important to modify them, which helps finds bugs, and doesn't increase the chance of OOB.

You made me realize that if we move to add OOB checks later, we'd want to change this code (as we could then add them even to initial contents). I added a TODO.

Reply:
Where do we add OOB checks in the fuzzer now?

Reply:
In `makePointer` (which emits a pointer for a load/store/etc.) if we don't allow OOB then we mask it to make sure it fits in memory.